parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -183,6 +183,12 @@ public static <K, V> ControllerEventMessage<K, V> of(WorkContainer<K, V> work) {
      */
     private final AtomicBoolean currentlyPollingWorkCompleteMailBox = new AtomicBoolean();
 
+    /**
+     * used for mark for interruption exclusion
+     */
+    private final AtomicBoolean awaitingSubmittedTaskComplete = new AtomicBoolean();
```

Comment:
```suggestion
    /**
     * Indicates state of waiting while in-flight messages complete processing on shutdown.
     * Used to prevent control thread interrupt due to wakeup logic on rebalances
     */
    private final AtomicBoolean awaitingInflightProcessingCompletionOnShutdown = new AtomicBoolean();
```

Reply:
fixed